--- a/smartd.8.in
+++ b/smartd.8.in
@@ -478,15 +478,6 @@ and stop it by using the command:
 .B /usr/local/etc/rc.d/init.d/smartd stop
 
 .fi
-If you want \fBsmartd\fP to start running whenever your machine is
-booted, this can be enabled by using the command:
-.nf
-.B /sbin/chkconfig --add smartd
-.fi
-and disabled using the command:
-.nf
-.B /sbin/chkconfig --del smartd
-.fi
 
 .\" DO NOT MODIFY THIS OR THE FOLLOWING TWO LINES. THIS MATERIAL
 .\" IS AUTOMATICALLY INCLUDED IN THE FILE smartd.conf.5
--- a/smartd.cpp
+++ b/smartd.cpp
@@ -1038,7 +1038,7 @@ static void MailWarning(const dev_config
              "  NIS domain: %s\n\n"
              "The following warning/error was logged by the smartd daemon:\n\n"
              "%s\n\n"
-             "For details see host's SYSLOG (default: /var/log/messages).\n\n"
+             "For details see host's SYSLOG (default: /var/log/syslog).\n\n"
              "%s%s%s",
 	     hostname, domainname, nisdomain, message, further, original, additional);
   exportenv(environ_strings[10], "SMARTD_FULLMESSAGE", fullmessage);
