--- a/cciss.cpp
+++ b/cciss.cpp
@@ -24,6 +24,10 @@
 #  define _HAVE_CCISS
 #elif defined(__FreeBSD_kernel__)
 #  include <endian.h>
+#  ifdef __GLIBC__
+#  include <bsd/sys/cdefs.h>
+#  include <stdint.h>
+#  endif
 #  include CISS_LOCATION
 #  define _HAVE_CCISS
 #endif
--- a/os_freebsd.cpp
+++ b/os_freebsd.cpp
@@ -1096,6 +1096,7 @@ protected:
 
 private:
   int m_disknum; ///< Disk number.
+  int m_encnum;  ///< Enclosure number.
 };
 
 
